File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        sum = 0
        res = float('inf')
        low = 0
        high = 0
        for high in range(len(nums)):
            sum += nums[high]
            while(sum >= target):
                length = high - low + 1
                res = min(length, res)
                sum -= nums[low]
                low += 1
        return res if res != float('inf') else 0
        

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int sum =0; int s =0;
        int min = Integer.MAX_VALUE;
        for(int i = 0; i< nums.length; i++){
            sum  += nums[i];
             while( sum >= target ){
                min  =  Math.min(min , i-s+1 );
                sum -= nums[s];
                s++;
            }
        }
         
        return (min == Integer.MAX_VALUE) ? 0: min;
        
    }
}